From 4d9252fd87d589cee8bbae74d7bf79cbd4d9fa1b Mon Sep 17 00:00:00 2001
From: Karl Williamson <khw@cpan.org>
Date: Thu, 29 Nov 2018 11:50:58 -0700
Subject: [PATCH 1/3] locale.c: Fix wrong scope of #if's

The function print_bytes_for_locale() should be defined if DEBUGGING;
prior to this commit it didn't get defined unless LC_COLLATE was
defined on the platform.
---
 locale.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/locale.c b/locale.c
index c32dd3f6ed..0c94042c55 100644
--- a/locale.c
+++ locale.c
@@ -4303,6 +4303,10 @@ S_print_collxfrm_input_and_return(pTHX_
     PerlIO_printf(Perl_debug_log, "'\n");
 }
 
+#  endif    /* DEBUGGING */
+#endif /* USE_LOCALE_COLLATE */
+#ifdef DEBUGGING
+
 STATIC void
 S_print_bytes_for_locale(pTHX_
                     const char * const s,
@@ -4338,8 +4342,7 @@ S_print_bytes_for_locale(pTHX_
     }
 }
 
-#  endif   /* #ifdef DEBUGGING */
-#endif /* USE_LOCALE_COLLATE */
+#endif   /* #ifdef DEBUGGING */
 
 #ifdef USE_LOCALE
 
-- 
2.19.1

